src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/AbstractRequestResolver.java

```diff
@@ -0,0 +1,50 @@
+package com.minecolonies.coremod.colony.requestsystem.resolvers;
+
+import com.minecolonies.api.colony.requestsystem.location.ILocation;
+import com.minecolonies.api.colony.requestsystem.resolver.IRequestResolver;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * ----------------------- Not Documented Object ---------------------
+ */
+public abstract class AbstractRequestResolver<Requested> implements IRequestResolver<Requested>
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00119)